iidm/iidm-mergingview/src/main/java/com/powsybl/iidm/mergingview/MergingView.java

```diff
@@ -275,422 +281,439 @@ public int getSubstationCount() {
     @Override
     public Substation getSubstation(final String id) {
         return getSubstationStream()
-                .filter(s -> id.compareTo(s.getId()) == 0)
+                .filter(s -> id.equals(s.getId()))
                 .findFirst()
                 .orElse(null);
     }
```

Comment:
Not sure this method is efficient: why don't use the index instead?

Reply:
yes, i changed in order to use NetworkIndex from each Network